### PR TITLE
Switch to ruff

### LINKED ISSUE #840: Switch to ruff for formatting
To keep us in line with NiPreps.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #840.

## Changes proposed in this pull request

- Swap out black for ruff.
- Ignore a few pesky style issues that I just didn't want to address yet.

### CODE REVIEW

File: tox.ini
Comment (mattcieslak) [inline tox.ini:1]: Do you know if this will interact with our deploy-to-pypi workflow on circleci?

File: qsiprep/utils/shm.py
Comment (mattcieslak) [inline qsiprep/utils/shm.py:888]:
```suggestion
       \rarrow L_{1,2} = \frac{-3 \pm \sqrt{9 - 8 (1-n)}}{2}
       \rarrow L{1,2} = \frac{-3 \pm \sqrt{1 + 8n}}{2}
```

File: qsiprep/utils/shm.py
Comment (mattcieslak) [inline qsiprep/utils/shm.py:937]:
```suggestion
        AP = \sum_{l=2,4,6,...}{\frac{1}{2l+1} \sum_{m=-l}^l{|a_{l,m}|^n}}
```

**Review by mattcieslak — APPROVED** [review]
only possible issue is the packaging for pypi with tox. minor changes to inline latex

File: docs/sphinxext/github_link.py
Comment (tsalo) [reply docs/sphinxext/github_link.py:2]: It means that the source code is from another project, but is copied into our source code rather than installing it with pip for example.

File: qsiprep/utils/shm.py
Comment (tsalo) [reply qsiprep/utils/shm.py:937]: The math isn't rendering correctly right now so I'd like to fix it before we merge. Maybe this'll address the problem but let's not merge before I have a chance to fix it.

File: tox.ini
Comment (tsalo) [reply tox.ini:1]: Sorry, I can't see what specifically you're flagging in this file. I'm not sure how your comment became outdated since I haven't touched this file in the last hour. GitHub might just be being terrible.

**Pull request merged**
